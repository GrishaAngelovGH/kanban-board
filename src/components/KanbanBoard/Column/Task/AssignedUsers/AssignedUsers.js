import { Link } from "react-router-dom"

import Button from "react-bootstrap/Button"

import Avatar from "components/Avatar"
import Tooltip from "components/Tooltip"

import boardRepository from "persistent/persistentKanbanBoardRepository"
import userRepository from "persistent/persistentUserRepository"

const AssignedUsers = ({ taskId, columnId, assignedIds, showToastWithMessage }) => {
  const users = userRepository.getUsers()

  return (
    <div className="mb-3 d-flex align-items-center flex-wrap">
      {
        assignedIds.map(v => {
          const user = users.find(u => u.id === v)

          return (
            <Tooltip key={v} label={user.name}>
              <Link to={`/tasks/${user.id}`} className="text-decoration-none">
                <Avatar user={user} size={35} />
              </Link>
            </Tooltip>
          )
        })
      }
      {
        assignedIds.length > 0 && (
          <>
            <Tooltip label="Click on a user to see their assigned tasks">
              <i className="m-1 bi bi-info fs-3 text-primary border border-3 border-primary rounded-circle d-flex bg-light"></i>
            </Tooltip>
            <Tooltip label="Remove all assignments">
              <Button
                variant="outline-danger"
                className="rounded-circle bi bi-x d-flex justify-content-center align-items-center"
                style={{ width: 35, height: 35 }}
                onClick={() => {
                  assignedIds.forEach(assignedId => {
                    boardRepository.removeAssignedUserFromTask(taskId, columnId, assignedId)
                    boardRepository.toggleTaskActiveStatus(taskId, columnId, false)
                  })
                  showToastWithMessage("All assigned users were successfully removed")
                }}
              >
              </Button>
            </Tooltip>
          </>
        )
      }
    </div>
  )
}

export default AssignedUsers