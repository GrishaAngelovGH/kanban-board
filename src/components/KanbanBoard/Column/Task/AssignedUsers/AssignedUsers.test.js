import { render } from "@testing-library/react"

import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom"

import AssignedUsers from "./AssignedUsers"

jest.mock('persistent/persistentUserRepository')

test("should render Task/AssignedUsers component", () => {
  const view = render(
    <Router>
      <Routes>
        <Route
          path="/"
          element={
            <AssignedUsers ids={{ assignedIds: [1, 2] }} />
          }
        />
      </Routes>
    </Router>
  )

  expect(view).toMatchSnapshot()
})