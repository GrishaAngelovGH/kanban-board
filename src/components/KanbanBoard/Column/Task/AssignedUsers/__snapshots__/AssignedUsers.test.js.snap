// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Task/AssignedUsers component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-3 d-flex align-items-center flex-wrap"
      >
        <a
          class="text-decoration-none"
          href="/tasks/1"
        >
          <img
            alt="user"
            class="rounded-circle m-1"
            src="image_1"
            width="35"
          />
        </a>
        <a
          class="text-decoration-none"
          href="/tasks/2"
        >
          <img
            alt="user"
            class="rounded-circle m-1"
            src="image_2"
            width="35"
          />
        </a>
        <i
          class="m-1 bi bi-info fs-3 text-primary border border-3 border-primary rounded-circle d-flex bg-light"
        />
        <button
          class="rounded-circle bi bi-x d-flex justify-content-center align-items-center btn btn-outline-danger"
          style="width: 35px; height: 35px;"
          type="button"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-3 d-flex align-items-center flex-wrap"
    >
      <a
        class="text-decoration-none"
        href="/tasks/1"
      >
        <img
          alt="user"
          class="rounded-circle m-1"
          src="image_1"
          width="35"
        />
      </a>
      <a
        class="text-decoration-none"
        href="/tasks/2"
      >
        <img
          alt="user"
          class="rounded-circle m-1"
          src="image_2"
          width="35"
        />
      </a>
      <i
        class="m-1 bi bi-info fs-3 text-primary border border-3 border-primary rounded-circle d-flex bg-light"
      />
      <button
        class="rounded-circle bi bi-x d-flex justify-content-center align-items-center btn btn-outline-danger"
        style="width: 35px; height: 35px;"
        type="button"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
